Engine/TransactionHandlers/BrokerageTransactionHandler.cs

```diff
@@ -1083,10 +1083,7 @@ private void HandleAccountChanged(AccountEvent account)
         private void HandlePositionAssigned(OrderEvent fill)
         {
             // informing user algorithm that option position has been assigned
-            if (fill.IsAssignment)
-            {
-                _algorithm.OnAssignmentOrderEvent(fill);
-            }
+            _algorithm.OnAssignmentOrderEvent(fill);
```

Comment:
Curious why we have a broker,transactionHandler & algorithm callback for option assignment but not for exercise. I gave the `OptionExerciseAssignRegressionAlgorithm` a test, and the event actually also seems to go through the `OnOrderEvent`:
```
OrderEvent Time: 12/25/2015 05:00:00 OrderID: 4 EventID: 2 Symbol: GOOG  151224C00745000 Status: Filled Quantity: 1 FillQuantity: 1 FillPrice: 0  Message: Automatic Assignment
Assignment Time: 12/25/2015 05:00:00 OrderID: 4 EventID: 2 Symbol: GOOG  151224C00745000 Status: Filled Quantity: 1 FillQuantity: 1 FillPrice: 0  Message: Option Expired - Automatic Assignment
```

Reply:
*Not really related to the objective of this PR changes, but the PR triggered me to look deeper 😁

Reply:
Both of those appear to be assignment events -- this change aimed at not raising the event for exercises. The tx handler was filtering out exercise events by checking `IsAssignment == false`